### PR TITLE
chore: add localhost to e2e testing

### PR DESCRIPTION
**Pull request by RickyLB**
**Description**:

**Related issue(s)**:
- https://github.com/hashgraph/hedera-sdk-rust/issues/735

Fixes #

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: src/account/account_info_flow.rs [hunk 1]
```diff
@@ -34,6 +34,12 @@ async fn query_pk(client: &Client, account_id: AccountId) -> crate::Result<Publi
 
     match key {
         Key::Single(it) => Ok(it),
+        // Key::KeyList(it) => {
+        //     let key = it.keys[0];
+
+        //     if let
+
+        // }
```
Comment (izik1) [inline src/account/account_info_flow.rs]:
```suggestion
```

File: src/client/mod.rs [hunk 2]
```diff
@@ -306,6 +309,17 @@ impl Client {
...
+    #[must_use]
+    pub fn for_localhost() -> Self {
+        let mut network: HashMap<String, AccountId> = HashMap::new();
+        network.insert(DEFAULT_LOCAL_NODE_ADDRESS.to_string(), AccountId::new(0, 0, 3));
+
+        let client = Client::for_network(network).unwrap();
+        client.set_mirror_network([DEFAULT_LOCAL_MIRROR_NODE_ADDRESS.to_string()]);
+        client
+    }
+
```
Comment (izik1) [inline src/client/mod.rs]: Please don't make this public

File: src/client/mod.rs [hunk 3]
```diff
@@ -352,6 +366,7 @@ impl Client {
             "mainnet" => Ok(Self::for_mainnet()),
             "testnet" => Ok(Self::for_testnet()),
             "previewnet" => Ok(Self::for_previewnet()),
+            "localhost" => Ok(Self::for_localhost()),
```
Comment (izik1) [inline src/client/mod.rs]: likewise

File: src/contract/contract_info.rs [hunk 4]
```diff
@@ -175,7 +175,9 @@ impl ToProtobuf for ContractInfo {
 #[cfg(test)]
 mod tests {
     use expect_test::expect;
-    use hedera_proto::services::{self,};
+    use hedera_proto::services::{
+        self,
+    };
```
Comment (izik1) [inline src/contract/contract_info.rs]:
```suggestion
    use hedera_proto::services;
```

File: src/contract/contract_log_info.rs [hunk 5]
```diff
@@ -70,7 +70,9 @@ impl ToProtobuf for ContractLogInfo {
 #[cfg(test)]
 mod tests {
     use expect_test::expect;
-    use hedera_proto::services::{self,};
+    use hedera_proto::services::{
+        self,
+    };
```
Comment (izik1) [inline src/contract/contract_log_info.rs]:
```suggestion
    use hedera_proto::services;
```

File: tests/e2e/common/mod.rs [hunk 6]
```diff
@@ -24,30 +24,37 @@ static CONFIG: Lazy<Config> = Lazy::new(Config::parse_env);
...
+        }
         Err(e) => {
             // to ensure we don't spam the logs with `Error creating client: ...`,
             // we just let an arbitrary thread win and log the "error".
             static LOGS_ONCE: AtomicBool = AtomicBool::new(false);
 
             // note: Relaxed is probably fine, AcqRel is *definitely* fine.
             if !LOGS_ONCE.swap(true, std::sync::atomic::Ordering::AcqRel) {
-                log::error!("Error creating client: {e}; creating one using `testnet`");
+                log::error!("Error creating client: {e}; creating one using `localhost`");
```
Comment (izik1) [inline tests/e2e/common/mod.rs]: testnet should still be the default, it's almost always going to be available, but local node might not

File: tests/e2e/common/mod.rs [hunk 7]
```diff
@@ -144,14 +152,17 @@ impl Config {
...
 
 pub(crate) fn setup_global() -> TestEnvironment {
     let _ = dotenvy::dotenv();
 
     let _ = env_logger::builder().parse_default_env().is_test(true).try_init();
 
-    TestEnvironment { config: &CONFIG, client: client() }
+    let (is_local, client) = client();
+
+    TestEnvironment { config: &CONFIG, client, is_local }
```
Comment (izik1) [inline tests/e2e/common/mod.rs]: Could `is_local` be on `config` rather than directly on `TestEnvironment`? That's what it's meant for

**Review by izik1 — CHANGES_REQUESTED** [review]

File: src/client/mod.rs (hunk 2 above)
Comment (RickyLB) [reply src/client/mod.rs:322]: @izik1 Do you mean only make it accessible to the testing environment?

File: tests/e2e/common/mod.rs [hunk 8]
```diff
@@ -131,13 +152,20 @@ impl Config {
     fn parse_env() -> Self {
         let network_name = dotenvy::var(keys::NETWORK).ok();
 
-        let network_name = network_name.map_or_else(|| Cow::Borrowed("testnet"), Cow::Owned);
+        let mut is_local = false;
+
+        // default tests to localhost
+        let network_name = network_name.map_or_else(|| Cow::Borrowed("localhost"), Cow::Owned);
```
Comment (izik1) [inline tests/e2e/common/mod.rs]: Missed a reversion of default?

**Review by izik1 — CHANGES_REQUESTED** [review]

**Review by izik1 — APPROVED** [review]

**Pull request merged**